src/update_feed.py

```diff
@@ -64,52 +68,76 @@ def __str__(self):
     r"(?P<major>\d+)\.?(?P<minor>\d+)*\.?(?P<patch>\d+)*(?P<label>\S*)"
 )
 
+GH_TOKEN = os.getenv("GITHUB_TOKEN", None)
 
-def normalize_version(version: str) -> SemVer:
+
+def normalize_version(version: str | SemVer) -> SemVer:
     if isinstance(version, SemVer):
         return version  # already normalized
     if len(version) == 0:
         return SemVer(0, 0, 0)
 
     result = re.search(SEMVER_PATTERN, version)
     if result is None:
-        logger.warning(f"failed to parse version '{version}'")
+        logger.debug(f"failed to parse version '{version}'")
         return SemVer()
 
     *nums, label = result.groups()
     major, minor, patch = [int(n) if n is not None else 0 for n in nums]
     return SemVer(major, minor, patch, label.strip("-."))
 
 
+@dataclass
+class Asset:
+    name: str
+    dl_url: str
+    created_at: str | None = None
+    content_type: str | None = None
+
+
+@dataclass
+class Release:
+    name: str
+    url: str
+    version: SemVer | None = None
+    date: datetime.datetime | None = None
+    tag: str | None = None
+    tag_as_variant: bool = False
+    asset_entries: list[dict] | None = None
```

Comment:
Would it make sense to specify a type of dict? For example, `list[dict[string, string]]` or is going to be arbitrary typed?

Reply:
Good point, this type hint is lacking. I'll add a proper type instead.